core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -64,6 +86,23 @@
 public class AlluxioMasterProcess implements MasterProcess {
   private static final Logger LOG = LoggerFactory.getLogger(AlluxioMasterProcess.class);
 
+  // Master metadata management.
```

Comment:
With all this new functionality, I think it might be time to split the `MetaMaster` off into its own master implementation, similar to `FileSystemMaster` or `BlockMaster`. Otherwise this class is getting very long and complex.

Reply:
Agree! To make AlluxioMasterProcess simple and have a separate MetaMaster. Could I have a separate PR to split AlluxioMasterProcess and MetaMaster? I think this part @apc999 @gpang may have different ideas.

Reply:
Yeah, having a "true" metamaster would be good.

Reply:
Separate PR is fine